src/main/java/com/snowflake/kafka/connector/internal/LoggerHandler.java

```diff
@@ -0,0 +1,122 @@
+package com.snowflake.kafka.connector.internal;
+
+
+import com.snowflake.kafka.connector.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
+
+public class LoggerHandler {
+  // static properties and methods
+  private static final UUID CORRELATION_ID_EMPTY = null;
+  private static final Logger META_LOGGER = LoggerFactory.getLogger(LoggerHandler.class.getName());
+  private static UUID LOGGER_CORRELATION_ID = CORRELATION_ID_EMPTY;
+
+  // should only be called on start
+  public static void setCorrelationUuid(UUID correlationId) {
+    LOGGER_CORRELATION_ID = correlationId;
+    META_LOGGER.info(Utils.formatLogMessage("Setting correlationId for all logging in this instance of Snowflake Kafka" +
```

Comment:
@sfc-gh-japatel would add correlationId at instance level be sufficient for the Fidelity case? IIRC, we had issues to figure out which partition or which thread was generating the log

Reply:
@sfc-gh-tzhang to clarify, is there one thread per partition?

Unfortunately I don't know how Java treats static properties/objects with threads, but I assume we may not be able to add thread level correlationIds for static classes. For non-static threads, it would be really easy to append a short additional hash to each logger with EnableLogging.java. I can make that change if needed

Reply:
Discussed offline, would like to add a task correlationId. jira: https://snowflakecomputing.atlassian.net/browse/SNOW-665587